ssh/tunnel.py

```diff
@@ -81,6 +82,9 @@ def tunnel(user: str, key_path: str, host: str, port: int=22):
     logger.debug('Closing SSH Tunnel: ' + ' '.join(close_tunnel))
     check_call(close_tunnel)
 
+    if os.path.exists(tunnel_socket.name):
+        os.remove(tunnel_socket.name)
```

Comment:
@lingmann , notice the tempfile `/teamcity/temp/buildTmp/tmp0ys0e50q` here, which is the NamedTemperoryFile (tunnel_socket.name) that is used with SSH tunnel command `-oControlPath`.

The theory is, when ssh exit happens, the ControlPath is cleaned up too. that's why the current Exception (harmless) is observed.


```
[11:13:37]	[Step 1/1] [2016-12-20 19:13:37,181|ssh.tunnel|DEBUG]: Closing SSH Tunnel: /usr/bin/ssh -oConnectTimeout=10 -oControlMaster=auto -oControlPath=/teamcity/temp/buildTmp/tmp0ys0e50q -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oBatchMode=yes -oPasswordAuthentication=no -p 22 -O exit core@35.156.217.233
[11:13:37]	[Step 1/1] Exit request sent.
[11:13:37]	[Step 1/1] Exception ignored in: <bound method _TemporaryFileCloser.__del__ of <tempfile._TemporaryFileCloser object at 0x7fccadf82438>>
[11:13:37]	[Step 1/1] Traceback (most recent call last):
[11:13:37]	[Step 1/1]   File "/usr/lib/python3.5/tempfile.py", line 589, in __del__
[11:13:37]	[Step 1/1]     self.close()
[11:13:37]	[Step 1/1]   File "/usr/lib/python3.5/tempfile.py", line 585, in close
[11:13:37]	[Step 1/1]     unlink(self.name)
[11:13:37]	[Step 1/1] FileNotFoundError: [Errno 2] No such file or directory: '/teamcity/temp/buildTmp/tmp0ys0e50q'
```

By checking and cleaning, I guess, it is just being proactive to verify if the file that we asked Tempfile module to not clean (delete=False), exists and if so, we clean it.